src/services/pocket-relayer.ts

```diff
@@ -652,7 +652,7 @@ export class PocketRelayer {
         blockchainEnforceResult && // Is this blockchain marked for result enforcement // and
         blockchainEnforceResult.toLowerCase() === 'json' && // the check is for JSON // and
         (!checkEnforcementJSON(relayResponse.payload) || // the relay response is not valid JSON // or
-          relayResponse.payload.startsWith('{"error"')) // the full payload is an error
+          (checkRelayError(relayResponse.payload) && !checkNodeError(relayResponse.payload))) // check if the payload indicates relay error, not a node error
```

Comment:
This logic confused me a good bit and I had to read it quite a few times. I think for me the hangup was the name

> checkNodeError

can we change that to

> isUserError

and the other one to

> isRelayError

Then it reads better to me, thanks. Then it will say

"Is this a relay error and not a user error?"  then it is a valid error, return it.

Otherwise it is user error and we should return that right away.